### PR TITLE
PAY-2391 Send data to CRM Topic

### PR DESCRIPTION
**Pull request by AbidemiAdio**

### CODE REVIEW

**Review by johnwatson484 — COMMENTED** [review]
Added some comments.

Summary is we don't need to handle retries because it's already covered and we can get the blob path from the response to publishing it to storage, rather than it need to be passed in through config.

**Comment by johnwatson484** [comment]
This line will return a blob upload response, which includes the full URL to storage.

https://github.com/DEFRA/ffc-pay-statement-generator/blob/55d439b22479083fcc909ce33d40c1950b58f56d/app/generator/publish.js#L21

File: docs/asyncapi.yaml [hunk 1]
```diff
@@ -15,6 +15,34 @@ components:
...
+                type: number
+                description: Firm Reference Number
+                minimum: 1000000000
+                maximum: 9999999999
+                example: 1234567890
+              blobUrl:
+                type: string
+                description: url address of statement file that has been published unto blob
+                example: 'https://devffcpayst1001.blob.core.windows.net/statements/outbound/FFC_PaymentStatement_SFI_2022_1234567890_2022090615023001.pdf'
+          type: 'uk.gov.pay.statement.crm'
```
Comment (johnwatson484) [inline docs/asyncapi.yaml:44]: Probably tweak the example to be `myBlobStorageAccount.blob.core` or something so it's not our actual account name.

File: test/unit/messaging/create-crm-message.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    })
+  }
+})
+
+jest.mock('../../../app/messaging/crm/crm-schema')
+const schema = require('../../../app/messaging/crm/crm-schema')
+
+const createCrmMessage = require('../../../app/messaging/crm/create-crm-message')
+const mockStatement = require('../../mocks/statement-data')
+const BLOB_URL = 'https://devffcpayst1001.blob.core.windows.net/statements/outbound/FFC_PaymentStatement_SFI_2022_1234567890_2022080515301012.pdf'
```
Comment (johnwatson484) [inline test/unit/messaging/create-crm-message.test.js:19]: Same comment, would amend the name

File: test/unit/messaging/send-crm-message.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    })
+  }
+})
+
+jest.mock('../../../app/messaging/crm/create-crm-message')
+const createCrmMessage = require('../../../app/messaging/crm/create-crm-message')
+
+const sendCrmMessage = require('../../../app/messaging/crm/send-crm-message')
+const mockStatement = require('../../mocks/statement-data')
+const BLOB_URL = 'https://devffcpayst1001.blob.core.windows.net/statements/outbound/FFC_PaymentStatement_SFI_2022_1234567890_2022080515301012.pdf'
```
Comment (johnwatson484) [inline test/unit/messaging/send-crm-message.test.js:19]: And again

File: app/messaging/crm/send-crm-message.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const { crmTopic } = require('../../config')
+const { MessageSender } = require('ffc-messaging')
+const createCrmMessage = require('./create-crm-message')
+
+const sendCrmMessage = async (statement, filename) => {
```
Comment (johnwatson484) [inline app/messaging/crm/send-crm-message.js:5]: Suggest renaming `filename` to `blobUrl` so it's clearer what it is.

**Review by johnwatson484 — COMMENTED** [review]
Just minor comments.

File: app/messaging/crm/send-crm-message.js (hunk 4 above)
Comment (AbidemiAdio) [reply app/messaging/crm/send-crm-message.js:5]: How did I not change this? copy and paste effect.

**Review by johnwatson484 — APPROVED** [review]
LGTM

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**